lib/jsonapi/plugs/underscore_parameters.ex

```diff
@@ -1,13 +1,25 @@
 defmodule JSONAPI.UnderscoreParameters do
   @moduledoc """
-  Takes dasherized JSON:API params and deserializes them to underscored params. Add
+  Takes dasherized JSON:API params and converts them to underscored params. Add
   this to your API's pipeline to aid in dealing with incoming parameters such as query
   params or data.
 
+  By default the newly underscored params will only replace the existing `params` field
+  of the `Plug.Conn` struct, but leave the `query_params` and `body_params` untouched.
+  If you are using the `JSONAPI.QueryParser` and need to also have the `query_params` on
+  the `Plug.Conn` updated, set the `replace_query_params` option to `true`.
```

Comment:
Hi!  So I'm trying to piece together the history here.

The original incarnation of the UnderscoreParameters was [this](https://github.com/beam-community/jsonapi/issues/145) - to stop dasherizing fields and work with the QueryParser.   cc @jherdman. 

The original documentation for that PR was....

```
  Given a request like:

       GET /example?filters[dog-breed]=Corgi
```
  
I suspect this was just an implementation oversight.  For incoming params, the benefit of the UnderscoreParameters plug was for convenience. I would assume replacing query-params also brings that convenience.  I get how this might be a breaking change for some rather than a bug fix though.  

We should fully expect it to be the default rather than a config option in the future.

Reply:
Definitely an oversight on my part.